gensim/models/doc2vec.py

```diff
@@ -585,18 +592,22 @@ def _vocab_from(self, documents, progress_per=10000):
                 interval_start = default_timer()
                 interval_count = total_words
             document_length = len(document.words)
+
             for tag in document.tags:
                 self.docvecs.note_doctag(tag, document_no, document_length)
+
             for word in document.words:
-                total_words += 1
-                if word in vocab:
-                    vocab[word].count += 1
-                else:
-                    vocab[word] = Vocab(count=1)
+                vocab[word] += 1
+
+            if self.max_vocab_size and len(vocab) > self.max_vocab_size:
+                total_words += utils.prune_vocab(vocab, min_reduce)
+                min_reduce += 1
```

Comment:
Constantly escalating `min_reduce` is likely overkill, discarding more words than strictly necessary. (I suspect it'd thus also increase the inaccuracy of final results – how different the N picked by this tend to be, compared to the true N most common words.)